### DISCUSSION TITLE
Support for 2FA

### DISCUSSION THREAD

**Discussion by BeauGiles** [pypolestar/polestar_api#80] [General]
Polestar is adding 2FA this month (January 2024), which will probably kill this Home Assistant integration again until there's an official API access method (or someone goes to the effort of forwarding/capturing the code sent via email).

https://www.polestar.com/au/polestar-support/polestar-id/

> Two-factor authentication adds an extra layer of security to your account. In addition to your Polestar ID username and password, you’ll need to enter a code (sent to you via email or SMS) to log in.
> 
> Two-factor authentication will be launched in January 2024. To prepare, please go to [polestarid.eu.polestar.com/Account/landing-page](https://polestarid.eu.polestar.com/PolestarLogin/login?resumePath=sdFIf4wmMo&client_id=pol1d0) and make sure that you:
> • can log in to your account.
> • have a valid mobile number or email address linked to your account.
> Note: Your Polestar ID username is the email address or mobile number that will receive the code when two-factor authentication launches.

**Comment by leeyuentuen** [comment]
let see when it has been added

**Comment by loebse** [comment]
I also just received an email from Polestar that they will start implementing the 2FA functionality starting January 2024. There is one detail that is interesting though: it says that they will enable it for "some services" soon. So, it will be very interesting what will happen 😉

**Comment by leeyuentuen** [comment]
this was my mail:

I can only implement once this has been activated.

Polestar-ID
--

Tweestapsverificatie wordt binnenkort geactiveerd
 
Voor extra veiligheid activeert Polestar vanaf januari 2024 tweestapsverificatie voor enkele van zijn services.Zorg ervoor dat het juiste e-mailadres of mobiele telefoonnummer aan je Polestar ID-account is gekoppeld. Als je e-mailadres of mobiele telefoonnummer onlangs is gewijzigd, moet je mogelijk je account eerst bijwerken. Klik hieronder voor meer informatie over welke stappen je dient te ondernemen.

![image](https://github.com/leeyuentuen/polestar_api/assets/1487966/b5f8b73d-4247-41a2-bd2f-867ef21b2918)
